Upgrade jsdom canvas

lightpriest opened the pull request:
Upgrades jsdom and canvas to the latest packages. This should make pdf-extractor easier to install since canvas >= 2 comes with pre-built binaries automatically.

Made the proper adjustments to domfacade to deal with the changes made to jsdom and canvas.

lightpriest committed:
Upgrade jsdom and canvas

This is to start using canvas prebuilt packages.

lightpriest committed:
Adjust domfacade to jsdom/canvas changes

Hikariii commented on lib/pdfjs-utils/domfacade.js:
```diff
@@ -10,7 +10,7 @@ const utils = require("jsdom/lib/jsdom/living/generated/utils");
 const HTMLElement = require("jsdom/lib/jsdom/living/generated/HTMLElement");
 const HTMLCanvasElement = require("jsdom/lib/jsdom/living/generated/HTMLCanvasElement");
-const Canvas  = require("jsdom/lib/jsdom/utils").Canvas;
+const { Canvas, Context2d } = require("jsdom/lib/jsdom/utils").Canvas;
```
Canvas 2 also export `Image` and `ImageData` in a new way.
This is exported here https://github.com/ScientaNL/pdf-extractor/blob/master/lib/pdfjs-utils/domfacade.js#L199-L200

```suggestion
const { Canvas, Context2d, Image, ImageData } = require("jsdom/lib/jsdom/utils").Canvas;
```

Hikariii replied:
Can you change lines 199 and 200 too with after importing them from canvas:
```
exports.Image = Image;
exports.ImageData = ImageData;
```

lightpriest replied:
Oops, missed that. Ran the examples on the README to make sure.

Hikariii reviewed (changes requested):
Thank you for your contribution.
There is still one change missing with the new canvas that is giving a new error:
Warning: Native JPEG decoding failed -- trying to recover: Image is not a constructor
If you can fix that i'll merge the change.

lightpriest committed:
Fix other Canvas based constructors

This is the same as Context2d. Image and ImageData are now exported
differently.

Hikariii commented:
Thank you very much!
